src/base/stream_entry.hpp

```diff
@@ -320,6 +320,15 @@ namespace cubstream
 
       void add_packable_entry (PO *entry)
       {
+#define MAX_PACKABLE_ENTRIES 1000
```

Comment:
This is a generic stream entry.
I think the packable entries limit and the action to take when this limit is reached should belong to specialization of stream_entry.

One solution could be to implement this only for online replication steam in :
  void
  log_generator::append_repl_object (replication_object &object)
  {
    m_stream_entry.add_packable_entry (&object);
   if (m_stream_entry.count_entries () >= MAX_PACKABLE_ENTRIES)  
       {
           (void) pack_stream_entry ();            // we may want to handle returned steam_position
       }
    er_log_repl_obj (&object, "log_generator::append_repl_object");
  }
 
The other would be to have in stream_entry class a data member m_max_packable_objects  and a std::function for the action to take when limit is reached. These would be set for each object of specializations of stream_entry.

Reply:
In my opinion, adding it to append_repl_object is easy and most clear option.

Reply:
Indeed, it looks good to move the code in append_repl_object rather than add_packable_entry. 

But, my concern is the size of stream entry. This was a quick fix, but probably we have to consider the memory size together with the number of objects. This will avoid stream unpack issue (that slow down the system when large memory is allocated). 

Also, another issue is on log consumer. We need to apply each stream entry before commit/abort. Otherwise, a lot of memory referred by many stream entries of same transactions, may slow down commit/abort of other transactions.

Even with all these modifications, the issue remains at unpack, that should not be so slow. I'm afraid also about a similar issue at pack when large memory is involved.

Reply:
I think you are right about number of objects not being enough. Total memory size may be more relevant here.

Maybe memory size can be used as limitation or maybe both criteria.

Reply:
CBRD-22851 should handle this case (size of objects)

Reply:
Both criteria must be used. For current patch, only the number of objects is considered.

As we discussed, total memory size will be fixed at 22581. However, splitting in several stream entries doesn't fix completely the issue (we may have a relatively big stream entry).

We need to open another issue "Slow unpack/pack for big stream entry". We can test and catch with various stream entry size like 1MB, 2MB... to notice where the serious time difference occurs.